### PR TITLE
adding changes: seperate admin ui to add xamcl policies

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js [hunk 1]
```diff
@@ -50,23 +51,55 @@ function loadAvailableJavaPolicies() {
...
 
-            //policyGroupIndexArray.push(obj.policyGroupId);
         }
     });
 
     //draw java policy items
     $("#javaPolicy_tbody").trigger("draw");
+
+    //update the url pattern wise policy group drop downs
+    $('.javaPolicy-opt-val').each(function () {
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: This code block does nothing, other than having a TODO comment. Remove this code block and add a descriptive TODO if it is really needed.

File: modules/jaggery-apps/app-workflow-admin/site/pages/index.jag [hunk 2]
```diff
@@ -16,6 +16,8 @@ middlePage="subscription-task";
 middlePage="signup-task";
 }else if(pageName=="publishapp"){
 middlePage="publishapp-task";
+}else if(pageName=="xacmlpolicy"){
+middlePage="xacmlpolicy-task";
```
Comment (rushmin) [inline modules/jaggery-apps/app-workflow-admin/site/pages/index.jag]: Indentation needed.

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+function savePolicyPartial() {
+
+    var policyPartial = $('#policy-content').val();
+    var policyPartialName = $('#policy-name').val();
+    var policyPartialDesc = $('#policy-desc').val();
+
+    var provider = "";
+    var isSharedPartial = true;
+    if (editedpolicyPartialId == 0) { //add
```
Comment (rushmin) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: LIne comment is not descriptive. Either remote it or add  a descriptive one.

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+
+        $.ajax({
+            async: false,
+            url: '/publisher/api/entitlement/get/apps/associated/to/xacml/policy/id/' + editedpolicyPartialId,
+            type: 'GET',
+            contentType: 'application/json',
+            dataType: 'json',
+            success: function (response) {
+
+                console.info(JSON.stringify(response));
```
Comment (rushmin) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: Remove this log.

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+                var apps = "";
+                if (response.length != 0) {
+                    // construct and show the  the warning message with app names which use this partial before update
+                    for (var i = 0; i < response.length; i++) {
+                        var j = i + 1;
```
Comment (rushmin) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: No need to use a another variable to store i + 1

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+            $('#policy-desc').val(obj.description);
+            $('#policy-content').val(obj.policyPartial);
+
+
+        }
+    });
+});
+
+
+//delete event
```
Comment (rushmin) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: Use proper commenting. e.g. Capitalizing

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js (hunk 1 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: this code was implemented during the parallel development with UI developer. This was fixed with a subsequent commit

File: modules/jaggery-apps/app-workflow-admin/site/pages/index.jag (hunk 2 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-workflow-admin/site/pages/index.jag]: fixed with commit# https://github.com/lahiru823/carbon-appmgt/commit/6560673570c990f0e0a73236c9b7de4a8d6a7160

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js (hunk 3 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: fixed with commit# https://github.com/lahiru823/carbon-appmgt/commit/d2a8716ceab0e122897d05abc37b35d464936aae

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js (hunk 4 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: This code was implemented during the parallel development with UI developer. This was fixed with a subsequent commit

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js (hunk 5 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: fixed with https://github.com/lahiru823/carbon-appmgt/commit/4f62c179404cabd6f5abf95a2d082e467235b86a

**Comment by lahiru823** [comment]
commit# https://github.com/lahiru823/carbon-appmgt/commit/28e79ae5fda095cde9082a5cab58531838733bac

File: modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+        contentType: 'application/x-www-form-urlencoded',
+        data: {"policyPartial": policyPartial},
+        success: onSuccess,
+        error: onError
+    });
+}
+
+
+function continueAddingEntitlementPolicyPartialAfterValidation(response) {
+    var response = JSON.parse(response);
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: Is it possible response to to be null here ? If so we have to handle it.

Comment (lahiru823) [inline modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js]: Here the response cannot be null. Its validated in upper layer.

**Pull request merged**